api/src/main/java/org/openmrs/util/OpenmrsUtil.java

```diff
@@ -1039,19 +1040,19 @@ public static String getApplicationDataDirectory() {
 		
 		if (filepath == null) {
 			if (OpenmrsConstants.UNIX_BASED_OPERATING_SYSTEM) {
-				filepath = System.getProperty("user.home") + File.separator + "." + openmrsDir;
+				filepath =  Paths.get(System.getProperty("user.home"), "." + openmrsDir).toString();
 				if (!canWrite(new File(filepath))) {
 					log.warn("Unable to write to users home dir, fallback to: "
 					        + OpenmrsConstants.APPLICATION_DATA_DIRECTORY_FALLBACK_UNIX);
-					filepath = OpenmrsConstants.APPLICATION_DATA_DIRECTORY_FALLBACK_UNIX + File.separator + openmrsDir;
+					filepath =Paths.get(OpenmrsConstants.APPLICATION_DATA_DIRECTORY_FALLBACK_UNIX, openmrsDir).toString();
```

Comment:
please format all your changes in this PR. I can see that some parts are missing whitespaces like in between the `=` in `=Paths.get(...`
to setup your IDE have a look at https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE